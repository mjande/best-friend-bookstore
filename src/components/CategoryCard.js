import { Link } from "react-router-dom";
import { kebabCaseToTitleCase } from "../helpers";
import "../styles/CategoryCard.css";

const CategoryCard = ({ name }) => {
  return (
    <div className={`category ${name}`}>
      <Link to={`/books/subject/${name}`} className="category-link">
        {kebabCaseToTitleCase(name)}
      </Link>
    </div>
  );
};

export default CategoryCard;
