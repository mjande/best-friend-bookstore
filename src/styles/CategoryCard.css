.category {
  border: solid;
  display: flex;
  justify-content: center;
  align-items: center;
}

.category-link {
  background-color: rgba(0, 0, 0, 0.7);
  border-radius: 0.5rem;
  padding: 0.5rem;
  color: white;
}

.category.mystery {
  background-image: url("../img/pexels-cottonbro-studio-8369520.jpg");
}

.category.romance {
  background-image: url("../img/pexels-picjumbocom-196664.jpg");
}

.category.science-fiction {
  background-image: url("../img/pexels-hristo-fidanov-1252890.jpg");
}

.category.fantasy {
  background-image: url("../img/pexels-craig-adderley-3359734.jpg");
  background-position: center;
  background-size: cover;
}
