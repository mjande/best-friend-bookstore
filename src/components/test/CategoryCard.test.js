import React from "react";
import { BrowserRouter } from "react-router-dom";
import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";

import CategoryCard from "../CategoryCard";

describe("CategoryCard", () => {
  it("renders CategoryCard with link to category", () => {
    const { container } = render(<CategoryCard name="test" />, {
      wrapper: BrowserRouter,
    });

    const LinkElement = screen.getByRole("link");
    expect(LinkElement).toBeInTheDocument();
    expect(LinkElement).toHaveTextContent("Test");
    expect(LinkElement).toHaveAttribute("href", "/books/subject/test");

    expect(container).toMatchSnapshot();
  });
});
