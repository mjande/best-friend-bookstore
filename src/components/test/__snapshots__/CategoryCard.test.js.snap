// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryCard renders CategoryCard with link to category 1`] = `
<div>
  <div
    class="category test"
  >
    <a
      class="category-link"
      href="/books/test"
    >
      Test
    </a>
  </div>
</div>
`;
